packages/edit-post/src/components/sidebar/plugin-document-setting-panel/index.js

```diff
@@ -92,8 +88,11 @@ const PluginDocumentSettingPanel = ( {
 	);
 	const { toggleEditorPanelOpened } = useDispatch( editPostStore );
 
-	if ( undefined === name ) {
-		warning( 'PluginDocumentSettingPanel requires a name property.' );
+	if ( ! name ) {
```

Comment:
Nit: if I decide to name my panel's name `0`, it will work alright, but still fail here. Should we be a bit more strict, and warn against it only when the `name` is `undefined`, `null`, or an empty string?

Reply:
It's probably better to avoid "falsy" values as a name and use something descriptive, IMO.

Reply:
Well, for the protocol, I agree it's better to avoid falsy values, no doubt about that. 

However, a numeric (and falsy in this case) value was still allowed before, though, and the bug this PR is referencing would not encounter if I were using numeric `name`s for my plugin document settings panels. We're also not explicitly saying that falsy names are not supported/recommended; we only say that a name should be provided in the error message. So if you insist to make that change and disallow falsy values, this is a breaking change and has to be communicated both in the error shown to the developer, and in a CHANGELOG entry.